tests/test_butler.py

```diff
@@ -482,6 +482,35 @@ def testConstructor(self):
         self.assertIsNone(butler2.run)
         self.assertIs(butler.datastore, butler2.datastore)
 
+        # Test that we can use an environment variable to find this
+        # repository.
+        butler_index = Config()
+        butler_index["label"] = self.tmpConfigFile
+        for suffix in (".yaml", ".json"):
+            # Ensure that the content differs so that we know that
+            # we aren't reusing the cache.
+            bad_label = f"s3://bucket/not_real{suffix}"
+            butler_index["bad_label"] = bad_label
+            with ButlerURI.temporary_uri(suffix=suffix) as temp_file:
+                butler_index.dumpToUri(temp_file)
+                with unittest.mock.patch.dict(os.environ, {"BUTLER_REPOSITORY_INDEX": str(temp_file)}):
+                    self.assertEqual(Butler.get_known_repos(), set(("label", "bad_label")))
+                    uri = Butler.get_repo_uri("bad_label")
+                    self.assertEqual(uri, ButlerURI(bad_label))
+                    uri = Butler.get_repo_uri("label")
+                    butler = Butler(uri, writeable=False)
+                    self.assertIsInstance(butler, Butler)
+                    with self.assertRaises(KeyError) as cm:
+                        Butler.get_repo_uri("missing")
+                    self.assertIn("not known to", str(cm.exception))
+        with unittest.mock.patch.dict(os.environ, {"BUTLER_REPOSITORY_INDEX": "file://not_found/x.yaml"}):
+            with self.assertRaises(FileNotFoundError):
+                Butler.get_repo_uri("label")
```

Comment:
Could easily check for empty set return from `get_known_repos()` here, if you keep that.